### PR TITLE
testing: fix flakiness

### PR DESCRIPTION
**Pull request by vgonkivs**
This PR aims to fix potential flakinesses in networking tests.

### CODE REVIEW

**Review by liamsi — APPROVED** [review]

**Review by distractedm1nd — APPROVED** [review]

**Review by renaynay — APPROVED** [review]
LGTM utACK

**Review by walldiss — APPROVED** [review]

File: service/share/testing.go [hunk 1]
```diff
@@ -199,9 +199,9 @@ func (dn *dagNet) Connect(peerA, peerB peer.ID) {
 // It does a hard disconnect, meaning that disconnected peers won't be able to reconnect on their own
 // but only with dagNet.Connect or dagNet.ConnectAll.
 func (dn *dagNet) Disconnect(peerA, peerB peer.ID) {
-	err := dn.net.DisconnectPeers(peerA, peerB)
+	err := dn.net.UnlinkPeers(peerA, peerB)
 	require.NoError(dn.t, err)
-	err = dn.net.UnlinkPeers(peerA, peerB)
```
Comment (Wondertan) [inline service/share/testing.go:204]: 👍🏻

File: node/tests/swamp/swamp.go [hunk 2]
```diff
@@ -296,3 +297,14 @@ func (s *Swamp) remove(rn *node.Node, sn []*node.Node) ([]*node.Node, error) {
...
+// Disconnect allows to break a connection between two peers without any possibility to re-establish it.
+// Order is very important here. We have to unlink peers first, and only after that call disconnect,
+// otherwise pubsub could re-establish the connection.
+func (s *Swamp) Disconnect(peerA, peerB peer.ID) error {
+	err := s.Network.UnlinkPeers(peerA, peerB)
+	if err != nil {
+		return err
+	}
+	return s.Network.DisconnectPeers(peerA, peerB)
+}
```
Comment (Wondertan) [inline node/tests/swamp/swamp.go:310]: Would be better to make rcv the T, so it checks for errors inside. This just generally makes tests cleaner.

File: node/tests/swamp/swamp.go [hunk 3]
```diff
@@ -296,3 +297,14 @@ func (s *Swamp) remove(rn *node.Node, sn []*node.Node) ([]*node.Node, error) {
 	}
 	return sn, nil
 }
+
+// Disconnect allows to break a connection between two peers without any possibility to re-establish it.
```
Comment (Wondertan) [inline node/tests/swamp/swamp.go:301]: We technically can reconnect them by linking and connecting peers back, and I suggest the addition of another symmetric `Connect` method that does this. The comment can mention that this is __hard__ disconnect which can reconnect with respective `Connect`, peers themselves would not be able to reconnect.

File: node/tests/swamp/swamp.go [hunk 4]
```diff
@@ -296,3 +297,14 @@ func (s *Swamp) remove(rn *node.Node, sn []*node.Node) ([]*node.Node, error) {
...
+// Order is very important here. We have to unlink peers first, and only after that call disconnect,
+// otherwise pubsub could re-establish the connection.
```
Comment (Wondertan) [inline node/tests/swamp/swamp.go:303]: Pubsub information is related to the place where the method is used, not on the Disconnect methods itself

File: node/tests/swamp/swamp.go (hunk 3 above)
Comment (vgonkivs) [reply node/tests/swamp/swamp.go:309]: added

**Review by Wondertan — APPROVED** [review]

**Review by Bidon15 — APPROVED** [review]

**Pull request merged**